core/src/main/scala/com/softwaremill/react/kafka_api_opt1/Example.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_opt1
+
+import scala.concurrent.Future
+import akka.Done
+import com.softwaremill.react.kafka_api_producer.Producer
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import akka.stream.scaladsl.Flow
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import akka.stream.ClosedShape
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.RunnableGraph
+import akka.NotUsed
+import akka.stream.scaladsl.Broadcast
+import akka.stream.scaladsl.Merge
+import akka.stream.scaladsl.Unzip
+import akka.stream.scaladsl.Zip
+
+trait Example {
+  implicit val ec = scala.concurrent.ExecutionContext.Implicits.global
+
+  def consumerProvider: () => KafkaConsumer[Array[Byte], String] = ???
+
+  def producerProvider: () => KafkaProducer[Array[Byte], String] = ???
+
+  class DB {
+    def save(offset: Long, data: String): Future[Done] = ???
+  }
+}
+
+// Consume messages and store a representation, including offset in DB
+object Example3a extends Example {
+  val db = new DB
+
+  // this possibly commits the offset before it's saved in db,
+  // but that's not a problem for externally stored offset
```

Comment:
But isn't it a possibility that:
 1. message is committed by Consumer.source
 2. DB storage call fails

Or did I misunderstand something here?